cdap-api-common/src/main/java/co/cask/cdap/api/data/schema/Schema.java

```diff
@@ -204,7 +229,38 @@ public static Schema of(Type type) {
     if (!type.isSimpleType()) {
       throw new IllegalArgumentException("Type " + type + " is not a simple type.");
     }
-    return new Schema(type, null, null, null, null, null, null, null);
+    return new Schema(type, null, null, null, null, null, null, null, null);
+  }
+
+  /**
+   * Creates a {@link Schema} for the given logical type. The logical type given must be a
+   * {@link Schema.LogicalType}.
+   * @param logicalType LogicalType of the schema to create.
+   * @return A {@link Schema} with the given logical type.
+   */
+  public static Schema of(LogicalType logicalType) {
+    Type type;
+    switch (logicalType) {
+      case DATE:
+        type = Type.INT;
+        break;
+      case TIMESTAMP_MILLIS:
+        type = Type.LONG;
+        break;
+      case TIMESTAMP_MICROS:
+        type = Type.LONG;
+        break;
+      case TIME_MILLIS:
+        type = Type.INT;
+        break;
+      case TIME_MICROS:
+        type = Type.LONG;
+        break;
+      default:
+        throw new IllegalArgumentException("LogicalType " + logicalType + " is not a supported logical type.");
```

Comment:
we usually use IllegalStateException for this, when there's a default after all possible enum values.

Reply:
Why is it a state? The `logicalType` is coming from the argument.